Tasks/Xcode/postxcode.ts

```diff
@@ -19,8 +19,12 @@ async function run() {
         
         if (publishResults) {
             if (!useXcpretty) {
-                tl.warning(tl.loc('UseXcprettyForTestPublishing'));
-            } else {
+                throw tl.loc('UseXcprettyForTestPublishing');
```

Comment:
When we `throw` we should use `Error` and `setResult` with `err.message` below.  When JavaScript built-ins throw, they will throw `Error`.

Reply:
Good point, for some reason all places in this file didn't use it. I will update.

Reply:
Actually in the async/await pattern used in the tests, it does not matter. If we update the throw to use `Error`, we have to update setResult to use `err.message`. So we will have to update everywhere or keep what we have.

Reply:
Ok, that's fine.  `Error` will get `toString`'ed to show its message anyway.

Reply:
I think it should throw an Error.  As @brcrista noted, anyone console.log(err) will get toString'd I believe